WordPress/gutenberg
# Try: Aggressive TinyMCE deprecation

## What?
This PR attempts to experiment with an aggressive strategy for deprecating TinyMCE. 

I'd love to get some feedback from everyone!

## Why?
The end goal is to fully deprecate the Classic block and all TinyMCE assets because they end up adding a lot of burden to the loading experience. With that approach, eventually, we could fully remove TinyMCE one day.

## How?
The PR suggests that we wrap the functionality in a Gutenberg Experiment so that everyone can enable it to opt-in.

The experiment disables loading any TinyMCE assets and the classic block by default.

Then, if TinyMCE usage is detected, it refreshes the page and adds an additional query argument. On the subsequent page load, all assets are loaded, classic block as well.

That way, a new installation, or an installation that doesn't utilize tinyMCE will never have to load its assets or see the Classic block anymore.

Furthermore, we detect if the post has a classic block in the content and make sure to load it in that case.

Finally, we detect that the user just inputs some raw HTML in their editor, in which case we alter the messaging to suggest the user either convert to an HTML block (from where they can convert to blocks) or refresh the page if they insist on using the Classic block.

## Testing Instructions
* Checkout the PR and build Gutenberg.
* Go to Experiments http://localhost:8888/wp-admin/admin.php?page=gutenberg-experiments.
* Enable the "Disable TinyMCE and Classic block" experiment and click "Save Changes".
* Open the post editor with a new post: http://localhost:8888/wp-admin/post-new.php
* Verify that you don't see the Classic editor in the inserter.
* Verify that you don't see any assets when you search for "tinymce" in the network tab of your browser dev tools (except for the tinymce-proxy.js file, which is expected as part of this experiment)
* Verify that there are no errors in the console. If you are seeing any errors in the console, verify you also see them in `trunk`.
* Head to `/lib/experimental/disable-tinymce.php` and uncomment the 8th line by removing the leading `//`.
* Open the post editor with a new post: http://localhost:8888/wp-admin/post-new.php
* Verify that the page gets refreshed shortly after loading.
* Verify that you now have the Classic block available and it works well when inserted. 
* Verify that after the new page is loaded, you now have the `requiresTinymce` query argument set.
* Disable the experiment from the Experiments page.
* Open the post editor again.
* Verify that Classic block is available and it works well. 
* Delete the query argument manually from your browser's URL address.
* Verify that Classic block is still available and it works well. 
* Start a new post, and use the code editor to write a sentence.
* Change back to the visual editor, and verify you can see a message like on the screenshot below.
* Save the post, refresh the page, and verify you can see the paragraph loaded within the Classic block.

### Testing Instructions for Keyboard
None

## Screenshots or screencast 

The message one sees when they edit their post in "code editor", they add some raw HTML and they switch back to the visual editor:

![Screenshot 2023-05-25 at 17 52 41](https://github.com/WordPress/gutenberg/assets/8436925/68a60b9e-a675-4cae-9cdd-2ab53bc99f1b)
